Quadratic Elem support in TriangulatorInterface

roystgnr opened the pull request:
This is enough to let us mix and match lower- and higher-order elements in boundary and hole definitions for the same triangulation, and get a triangulation output that matches the higher-order midpoint nodes where they're available.

This should be everything we need on the libMesh end to support what I called "level 1" in https://github.com/libMesh/libmesh/pull/3811

roystgnr commented:
Broken in parallel, huh?

jwpeterson reviewed (approved):
Looks good to me aside from whatever parallel issue you are tracking down

roystgnr commented:
I rebased so it's hard to see, but the parallel fix was one line; I forgot that I had split a broadcast() into two places and when I had to add a `broadcast(_midpoints)` I forgot to add it to both.

Kind of an awkward code decision, actually.  I did:
```
if (pid != 0)
{
  broadcast(foo);
  return();
}

hundreds_of_lines();
of_pid_0_stuff();

broadcast(foo);
```

... pretty much solely because I didn't want to indent `hundreds_of_lines()` inside an `if (pid == 0)` block or create a separate `hundreds_of_lines_of_pid_0_stuff()` function.  I think my subconscious heuristic might be "code with smaller `wc -c` output is better" and I think it might have failed me here...

roystgnr commented:
Okay; I'll merge this once "Test 32bit" is done so we can get into the submodule update ASAP.

roystgnr commented:
It's done with unit tests; the examples don't cover this code.
